src/main/scala/Cqrs/Aggregate.scala

```diff
@@ -23,9 +28,9 @@ object Aggregate {
   final case class VersionedEvents[E](version: Int, events: List[E])
 
   sealed trait EventDatabaseOp[E, A]
-  final case class ReadAggregateExistence[E](id: AggregateId) extends EventDatabaseOp[E, Error Xor Boolean]
-  final case class ReadAggregate[E](id: AggregateId, fromVersion: Int) extends EventDatabaseOp[E, Error Xor List[VersionedEvents[E]]]
-  final case class AppendAggregateEvents[E](id: AggregateId, events: VersionedEvents[E]) extends EventDatabaseOp[E, Error Xor Unit]
+  final case class ReadAggregateExistence[E](tag: Tag, id: AggregateId) extends EventDatabaseOp[E, Error Xor Boolean]
```

Comment:
The semantics of the tags in this PR are a bit different from akka - its more like aggregate class - eg all counters, or all doors. Checking if exists also needs it in this case - "is there a counter (by tag) identified by this id". 

Projections tell what classes / tags of aggregates they are interested in and receive all aggregates within them